src/quacc/recipes/newtonnet/ts.py

```diff
@@ -293,3 +307,250 @@ def _get_hessian(atoms: Atoms) -> NDArray:
+
+    # Run IRC
+    irc_summary = strip_decorator(irc_job)(
+        atoms, direction=direction, run_freq=False, **irc_job_kwargs
+    )
+
+    # Run opt
+    relax_summary = strip_decorator(relax_job)(irc_summary["atoms"], **relax_job_kwargs)
+
+    # Run frequency
+    freq_summary = (
+        strip_decorator(freq_job)(relax_summary["atoms"], **freq_job_kwargs)
+        if run_freq
+        else None
+    )
+    relax_summary["freq_job"] = freq_summary
+    relax_summary["irc_job"] = irc_summary
+
+    return relax_summary
+"""
+
+
+def sella_wrapper(
+    atoms_object,
+    traj_file=None,
+    sella_order=0,
+    use_internal=True,
+    traj_log_interval=2,
+    fmax_cutoff=1e-3,
+    max_steps=1000,
+):
+    if traj_file:
+        traj = Trajectory(traj_file, "w", atoms_object)
+    qn = Sella(atoms_object, order=sella_order, internal=use_internal)
+    if traj_file:
+        qn.attach(traj.write, interval=traj_log_interval)
+    qn.run(fmax=fmax_cutoff, steps=max_steps)
+    if traj_file:
+        traj.close()
+
+
+def geodesic_interpolate_wrapper(
+    r_p_atoms: Atoms,
+    nimages: int = 17,
+    sweep: bool | None = None,
+    output: str = "interpolated.xyz",
+    tol: float = 2e-3,
+    maxiter: int = 15,
+    microiter: int = 20,
+    scaling: float = 1.7,
+    friction: float = 1e-2,
+    dist_cutoff: float = 3,
+    save_raw: str | None = None,
+):
+    """
+    Interpolates between two geometries and optimizes the path.
+
+    Parameters:
+    filename (str): XYZ file containing geometries.
+    nimages (int): Number of images. Default is 17.
+    sweep (bool): Sweep across the path optimizing one image at a time.
+                  Default is to perform sweeping updates if there are more than 35 atoms.
+    output (str): Output filename. Default is "interpolated.xyz".
+    tol (float): Convergence tolerance. Default is 2e-3.
+    maxiter (int): Maximum number of minimization iterations. Default is 15.
+    microiter (int): Maximum number of micro iterations for sweeping algorithm. Default is 20.
+    scaling (float): Exponential parameter for Morse potential. Default is 1.7.
+    friction (float): Size of friction term used to prevent very large change of geometry. Default is 1e-2.
+    dist_cutoff (float): Cut-off value for the distance between a pair of atoms to be included in the coordinate system. Default is 3.
+    save_raw (str): When specified, save the raw path after bisections but before smoothing. Default is None.
+    """
+    # Read the initial geometries.
+    symbols = r_p_atoms[0].get_chemical_symbols()
+
+    X = [conf.get_positions() for conf in r_p_atoms]
+
+    if len(X) < 2:
+        raise ValueError("Need at least two initial geometries.")
+
+    # First redistribute number of images. Perform interpolation if too few and subsampling if too many images are given
+    raw = redistribute(symbols, X, nimages, tol=tol * 5)
+    if save_raw is not None:
+        write_xyz(save_raw, symbols, raw)
```

Comment:
In quacc, we never write out files to the current working directory because that is not necessarily the same as the directory where the calculations are run. We will need to revisit this to ensure it is written to the proper directory.